crates/next-dev/tests/integration/next/image/basic/pages/index.js

```diff
@@ -0,0 +1,50 @@
+import Image from "next/image";
+import img from "../public/triangle-black.png";
+import { useEffect } from "react";
+
+import { Deferred } from "@turbo/pack-test-harness/deferred";
+
+let testResult = new Deferred();
+
+export default function Home() {
+  useEffect(() => {
+    // Only run on client
+    import("@turbo/pack-test-harness").then(runTests);
```

Comment:
We need a little orchestration in order to only import the jest harness in the client side. Because dynamic import can only happen async, we can't sync provide the jest test suite or the `__jest__.run()` function. So we require exposing the `waitForTests` global function so that the rust runner knows how to wait for the value.

If the `waitForTests` isn't expose, it assumes this is a sync test suite (that could sync import the test harness) and just runs.

Reply:
The source code of the test case shouldn't need to contain these test suite things...

But either way it could be just a `import "@turbo/pack-test-harness"` and the package can be made isomophic e. g. by using `"exports": { "node": "...", ... }`

Extra points if the test cases run in node.js too, but that's not important for now. It could be just `function it() { /* ignore */ }`

Reply:
The tests themselves still require React to render. I want to avoid making this a testing PR and keep it focused on `next/image`.